fe/fe-core/src/main/java/com/starrocks/mysql/MysqlProto.java

```diff
@@ -156,6 +157,15 @@ public static NegotiateResult negotiate(ConnectContext context) throws IOExcepti
             if (authPacket == null) {
                 return new NegotiateResult(null, NegotiateState.READ_SSL_AUTH_PKG_FAILED);
             }
+        } else if (Config.ssl_force_secure_transport) {
+            if (!NetUtils.isIPLocalAddress(context.getRemoteIP())) {
```

Comment:
Just check if it is 127.0.0.1. Like mysql.

Reply:
@gengjun-git, updated the PR to check for "127.0.0.1", please check. qq: Should we check if it’s a loopback address instead?

Reply:
Isn't 127.0.0.1 the loopback address? Are there other loopback addresses? You mean localhost or ipv6?

Reply:
Yes, 127.0.0.1 is a loopback address. However, as addresses in the range 127.x.x.x are also loopback addresses should we consider range and IPv6 loopback addresses as well?

Reply:
Don't need it now, for complex networks

Reply:
fyi for @kevincai